src/GovUk.Education.ExploreEducationStatistics.Data.Processor/FailedImportsHandler.cs

```diff
@@ -0,0 +1,98 @@
+    {
+        public static void CheckIncompleteImports()
+        {
+            var tblStorageAccount = CloudStorageAccount.Parse(ConnectionUtils.GetAzureStorageConnectionString("CoreStorage"));
+            var storageAccount = Microsoft.Azure.Storage.CloudStorageAccount.Parse(ConnectionUtils.GetAzureStorageConnectionString("CoreStorage"));
+            var tClient = tblStorageAccount.CreateCloudTableClient();
+            var qClient = storageAccount.CreateCloudQueueClient();
+            var aQueue = qClient.GetQueueReference("imports-available");
+            var pQueue = qClient.GetQueueReference("imports-pending");
+            var table = tClient.GetTableReference("imports");
+
+            aQueue.CreateIfNotExists();
+            table.CreateIfNotExists();
+
+            TableContinuationToken token = null;
+            do
+            {
+                var resultSegment =
+                    table.ExecuteQuerySegmentedAsync(BuildQuery(), token).Result;
+                token = resultSegment.ContinuationToken;
+
+                foreach (var entity in resultSegment.Results)
+                {
+                    var lastBatch = ImportStatusService.GetNumBatchesComplete(entity);
+                    if (entity.NumBatches != 1 && entity.NumBatches == lastBatch)
+                    {
+                        continue;
+                    }
+
+                    // Older entries will not have this recover mechanism so have to ignore 
+                    if (entity.Message != null)
+                    {
+                        Console.Out.WriteLine($"Recovering {entity.PartitionKey} : {entity.RowKey}");
+
+                        // If batch was not split then just processing again by adding to pending queue
+                        if (entity.NumBatches == 1 || lastBatch == 0)
+                        {
+                            pQueue.AddMessage(new CloudQueueMessage(entity.Message));
+                        }
+                        else
+                            // Add the batch that failed on to the queue is missing
+                        {
+                            aQueue.AddMessage(new CloudQueueMessage(BuildMessage(
+                                JsonConvert.DeserializeObject<ImportMessage>(entity.Message), 
+                                lastBatch+1)));
+                        }
+                    }
+                    else
+                    {
+                        Console.Out.WriteLine($"No message stored for import - unable to recover import " +
+                                              $"{entity.PartitionKey} : {entity.RowKey}");
+                    }
+                }
+            } while (token != null);
+        }
+        
+        private static TableQuery<DatafileImport> BuildQuery()
+        {
+            var f1 = TableQuery.GenerateFilterCondition("Status", 
+                QueryComparisons.Equal, IStatus.RUNNING_PHASE_1.GetEnumValue());
```

Comment:
I'm supprised these conditions are working because GetEnumValue is an extension method that we wrote for enums which expects them to be annotated with `EnumLabelValueAttribute` and gets the value attribute of this annotation. IStatus hasn't got any EnumLabelValueAttribute. I think in that exception it calls ToString which would probably be "RUNNING_PHASE_1".

If the intention is to compare the int values, just cast it as in int since enum's are already ints.

Reply:
Yeah worked OK but I've changed to toString to make clearer